### ISSUE TITLE
Support for multiple result sets

### ISSUE THREAD

**Issue by ekstrom** [livehelpnow/tds#108]
It would be useful to have support for multiple result sets.

Example code:

```elixir
query = "SELECT 1 AS 'first_result_set'; SELECT 2 AS 'second_result_set';"
Tds.query!(pid, query, [])
```

Expected result:

```elixir
[%Tds.Result{columns: ["first_result_set"], num_rows: 1, rows: [[1]]}, %Tds.Result{columns: ["second_result_set"], num_rows: 1, rows: [[2]]}]
```

Actual result:

```elixir
%Tds.Result{columns: ["first_result_set"], num_rows: 1, rows: [[1]]}
```

**Comment by mjaric** [comment]
Last year I refactored decoder part, it reads and keeps [all results](https://github.com/livehelpnow/tds/blob/master/lib/tds/messages.ex#L169) until it must hadoff result to user code. Reason is that ecto expects only single result, so `Tds.Protocol` pulls out last result [here](https://github.com/livehelpnow/tds/blob/master/lib/tds/protocol.ex#L796). 

Since ecto adapter does not use `Tds.query/3` nor `Tds.query/4` it is possible to pass additional flag in options that should tell internally Tds.Protocol module to rather return all results. For instance, [here](https://github.com/livehelpnow/tds/blob/master/lib/tds.ex#L14) it could be something like:

```elixir
  def query(pid, statement, params, opts \\ []) do
    query = %Query{statement: statement}
    opts = 
      opts
      |> Keyword.put_new(:parameters, params)
      |> Keyword.put(:resultset, true) # <-----   THE FLAG!

    case DBConnection.prepare_execute(pid, query, params, opts) do
      {:ok, _query, result} -> {:ok, result}
      {:error, err} -> {:error, err}
    end
  end
```

Then it could pack among all results additional info, like messages server returned (info, error ...), num_results, and result set count. 

It is worth mentioning that results are in reverse order so they need to be `Enum.reverse`d before they are handed to user code.

**Comment by mjaric** [comment]
One more not, since unit tests also expect single Tds.Result, all must be refactored to pass with this change

**Comment by josevalim** [comment]
Perhaps the best option is to introduce a separate function, such as `query_multi` or `query_many` or similar. So users can explicitly opt-in into it.

**Comment by mjaric** [comment]
@ekstrom  BTW, please point your fork to tds master and run test, but there shouldn't be anything broken

**Comment by ekstrom** [comment]
Tested it and it works as expected. Thanks for the good work :smile:

**Comment by mjaric** [comment]
Released in v2.1.1. 

Thank you!
